components/eamxx/src/physics/cosp/cosp_functions.hpp

```diff
@@ -0,0 +1,96 @@
+#ifndef SCREAM_COSP_FUNCTIONS_HPP
+#define SCREAM_COSP_FUNCTIONS_HPP
+#include "share/scream_types.hpp"
+using scream::Real;
+extern "C" void cosp_c2f_init(int ncol, int nsubcol, int nlay);
+extern "C" void cosp_c2f_final();
+extern "C" void cosp_c2f_run(int ncol, int nsubcol, int nlay, int ntau, int nctp,
+    Real emsfc_lw, Real* sunlit, Real* skt,
+    Real* T_mid, Real* p_mid, Real* p_int, Real* qv,
+    Real* cldfrac, Real* reff_qc, Real* reff_qi, Real* dtau067, Real* dtau105,
+    Real* isccp_cldtot, Real* isccp_ctptau);
+
+namespace scream {
+
+    namespace CospFunc {
+        using lview_host_1d = typename ekat::KokkosTypes<HostDevice>::template lview<Real*  >;
+        using lview_host_2d = typename ekat::KokkosTypes<HostDevice>::template lview<Real** >;
+        using lview_host_3d = typename ekat::KokkosTypes<HostDevice>::template lview<Real***>;
+        template <typename S>
+        using view_1d = typename ekat::KokkosTypes<HostDevice>::template view_1d<S>;
+        template <typename S>
+        using view_2d = typename ekat::KokkosTypes<HostDevice>::template view_2d<S>;
+        template <typename S>
+        using view_3d = typename ekat::KokkosTypes<HostDevice>::template view_3d<S>;
+
+        void initialize(int ncol, int nsubcol, int nlay) {
+            cosp_c2f_init(ncol, nsubcol, nlay);
```

Comment:
If you provide an impl for initialize/finalize, then
1. you don't need the cosp_functions.cpp file anymore (otherwise you have multiple def of the same fcn within each translation unit)
2. you need to make these functions inline (otherwise you will have multiple def of the same fcn across all translation units)

Reply:
Yeah, the cpp source was already not being used. I added the inline specifier to the function decls, please let me know if this is not what you meant here.